src/libcollections/vec_deque.rs

```diff
@@ -2158,6 +2158,18 @@ impl<A: PartialEq> PartialEq for VecDeque<A> {
 #[stable(feature = "rust1", since = "1.0.0")]
 impl<A: Eq> Eq for VecDeque<A> {}
 
+#[stable(feature = "vec-deque-partial-eq-slice", since = "")]
+impl<'a, A: PartialEq> PartialEq<&'a [A]> for VecDeque<A> {
```

Comment:
Looks like one would have to write `assert!(d == *&v[..]);`, which seems less than ideal.